plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergMetadata.java

```diff
@@ -1118,7 +1118,7 @@ private static IcebergColumnHandle createProjectedColumnHandle(IcebergColumnHand
                 column.getBaseType(),
                 fullPath.build(),
                 projectedColumnType,
-                Optional.empty());
+                column.getComment());
```

Comment:
I'd rather avoid unnecessary projected column creation itself if there're no indices. That should fix this issue too.

Reply:
This isn't exactly correct. The `IcebergColumnHandle` referencing a field in a Row should have no comment field, as the Row's field itself has no comment.

Revert.